build-scripts/multidirectory.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/sh
+# Creates/removes/updates branches according the `jobs/` directory on the
+# master branch.  Local files and branches will be modified, so this script
+# should be executed on a clean, dedicated clone.
+set -eu
+
+set_difference() {
+    printf '%s\n%s\n%s\n' "$1" "$1" "$2" | sort | uniq -u
+}
+
+list_branches() {
+    git branch --list --remotes 'origin/*' \
```

Comment:
The basic review I have for you does not change. Get general output from `git branch` and use a `for` loop to cleanly and iteratively apply filters. Don't make someone decode complicated `awk` expressions (that's two programming languages you need to know now, not just one!).